service_capacity_modeling/models/org/netflix/ddb.py

```diff
@@ -0,0 +1,584 @@
+        total_annual_data_transfer_cost=annual_transfer_cost,
+    )
+
+
+def _plan_backup(
+    context: RegionContext,
+    desires: CapacityDesires,
+    extra_model_arguments: Dict[str, Any],
+) -> _DataBackupPlan:
+    number_of_regions = _get_num_regions(extra_model_arguments)
+    dynamo_backup_continuous = _get_dynamo_backup(context)
+    annual_pitr_cost = (
+        math.ceil(desires.data_shape.estimated_state_size_gib.mid)
+        * dynamo_backup_continuous.annual_cost_per_gib[0][1]
+    )
+    # normalizing the cost as pitr is not charged per region
+    annual_pitr_cost_normalized = round((annual_pitr_cost / number_of_regions), 2)
+    return _DataBackupPlan(
+        total_backup_data_storage_gib=round(
+            desires.data_shape.estimated_state_size_gib.mid, 2
+        ),
+        total_annual_backup_cost=annual_pitr_cost_normalized,
+    )
+
+
+class NflxDynamoDBCapacityModel(CapacityModel):
+    @staticmethod
+    def capacity_plan(
+        instance: Instance,
+        drive: Drive,
+        context: RegionContext,
+        desires: CapacityDesires,
+        extra_model_arguments: Dict[str, Any],
+    ) -> Optional[CapacityPlan]:
+        write_plan = _plan_writes(context, desires, extra_model_arguments)
+        read_plan = _plan_reads(context, desires, extra_model_arguments)
+        storage_plan = _plan_storage(context, desires, extra_model_arguments)
+        backup_plan = _plan_backup(context, desires, extra_model_arguments)
+        data_transfer_plan = _plan_data_transfer(
+            context, desires, extra_model_arguments
+        )
+        requirement_context = {
+            "read_capacity_units": read_plan.read_capacity_units,
+            "write_capacity_units": write_plan.write_capacity_units,
+            "data_transfer_gib": data_transfer_plan.total_data_transfer_gib,
+        }
+        requirement_context[
+            "replicated_write_capacity_units"
+        ] = write_plan.replicated_write_capacity_units
+        requirement = CapacityRequirement(
+            requirement_type="dynamo-regional",
+            core_reference_ghz=0,
+            cpu_cores=certain_int(0),
+            disk_gib=certain_float(storage_plan.total_data_storage_gib),
+            context=requirement_context,
+        )
+        dynamo_services = [
+            ServiceCapacity(
+                service_type="dynamo.standard",
+                annual_cost=(
+                    write_plan.total_annual_write_cost
+                    + read_plan.total_annual_read_cost
+                    + storage_plan.total_annual_data_storage_cost
+                    + data_transfer_plan.total_annual_data_transfer_cost
+                ),
+                service_params={
+                    "read_capacity_units": read_plan.read_capacity_units,
+                    "write_capacity_units": (
+                        write_plan.write_capacity_units
+                        + write_plan.replicated_write_capacity_units
```

Comment:
Isn't there a storage as well?

Reply:
for creating ddb table, storage is not one of the inputs

Reply:
Yeah but that context is useful to have in the caller (what context was originally for heh)?

Reply:
CapacityRequirement(
            requirement_type="dynamo-regional",
            core_reference_ghz=0,
            cpu_cores=certain_int(0),
            disk_gib=certain_float(storage_plan.total_data_storage_gib),
            context=requirement_context,
        )